File: port_js/org/kbinani/ByRef.js
/*
 * ByRef.js
 * Copyright (C) 2010 kbinani
 *
 * This file is part of org.kbinani.
 *
 * org.kbinani is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * org.kbinani is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
if( org == undefined ) var org = {};
if( org.kbinani == undefined ) org.kbinani = {};
if( org.kbinani.ByRef == undefined ){

    org.kbinani.ByRef = function(){
        this.value = null;
        if( arguments.length == 1 ){
            this._init_1( arguments[0] );
        }
    };

    org.kbinani.ByRef.prototype = {
        _init_1 : function( value ){
            this.value = value;
            return this;
        },
    };

}

File: __DRAFT__org.kbinani.vsq/ByRef.cs
/*
 * ByRef.cs
 * Copyright (C) 2009-2010 kbinani
 *
 * This file is part of org.kbinani.vsq.
 *
 * org.kbinani.vsq is free software; you can redistribute it and/or
 * modify it under the terms of the BSD License.
 *
 * org.kbinani.vsq is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
namespace org {
    namespace kbinani {
        namespace vsq {

            public class ByRef<T> {
                public T value;

                public ByRef() {
                }

                public ByRef( T value_ ) {
                    value = value_;
                }
            }

        }
    }
}
